webcompat/static/js/lib/bugform.js

```diff
@@ -182,18 +193,28 @@ function BugForm() {
     img.src = dataURI;
   };
 
+  // Is the user trying to report a site against webcompat.com itself?
+  this.isSelfReport = function() {
+    var url = location.href.match(this.urlParamRegExp);
+    if (url !== null) {
+      if (_.includes(decodeURIComponent(url[0]), location.origin)) {
```

Comment:
What exactly is checked here?

If the following URL is given `https://www.webcompat.com/?url=https://www.webcompat.com`, then `url` would contain `["url=https://www.webcompat.com", "https://www.webcompat.com"]`.
So url[0] is `"url=https://www.webcompat.com"`.
`location.origin` is `"https://www.webcompat.com"`.
`_.includes(decodeURIComponent(url[0]), location.origin)` is true in this case.

What if the user visits using this URL?
`https://www.webcompat.com/?url=https://webcompat.com`

The check would fail, because `"url=https://webcompat.com"` does not contain `"https://www.webcompat.com"`.

@miketaylr Any chance to make this failsafe?

Reply:
> What exactly is checked here?

We check that the origin is somewhere inside the decoded `url` param. And yeah, in theory someone could visit with `https://www.webcompat.com/?url=https://webcompat.com`, though the likelihood is probably near zero. But let's fix it. :)

Reply:
After having another thought on this, I guess this is probably too complicated to solve with too few outcomes. If that's doing problems in the future we should consider fixing it then.